File: bindings/java-ffm/src/main/java/org/maplibre/nativeffi/error/WrongThreadException.java
package org.maplibre.nativeffi.error;

/** Thrown when an owner-thread-affine native handle is used from the wrong thread. */
public final class WrongThreadException extends MaplibreException {
  public WrongThreadException(int nativeStatusCode, String diagnostic) {
    super(MaplibreStatus.WRONG_THREAD, nativeStatusCode, diagnostic);
  }
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/error/WrongThreadException.kt
package org.maplibre.nativeffi.error

/** Error for owner-thread-affine native handles called from the wrong thread. */
public class WrongThreadException(nativeStatusCode: Int, diagnostic: String = "") :
  MaplibreException(MaplibreStatus.WRONG_THREAD, nativeStatusCode, diagnostic)
